### PR TITLE
Handling deletion of Port Channel before deletion of its members

### PR DESCRIPTION
**Pull request by madhanmellanox**
- What I did
Handled configuration code to block deletion of Port channel when user tries to delete a Port channel before deleting its members.

- How I did it
In the sonic-utilities/config/main.py, where the removal of Port channel configuration is handled, I added a check whether the Port channel to be deleted has any members. If members exist, I will block the configuration of deleting the port channel.

- How to verify it
Create a Portchannel. Add 2 switch ports to it as members. Then try to delete the port channel. Now, the deletion of port channel should be blocked by the config since it has port channel members still existing.

- Previous command output (if the output of a command-line utility has changed)
```
admin@r-qa-sw-eth-53:/tmp$ sudo config portchannel add portchannel100
admin@r-qa-sw-eth-53:/tmp$ sudo config portchannel member add portchannel100 Ethernet4
admin@r-qa-sw-eth-53:/tmp$ sudo config portchannel member add portchannel100 Ethernet8
admin@r-qa-sw-eth-53:/tmp$ show interfaces portchannel
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev        Protocol     Ports
-----  --------------  -----------  -------------------------
  100  PortChannel100  LACP(A)(Dw)  Ethernet8(D) Ethernet4(D)
admin@r-qa-sw-eth-53:/tmp$ sudo config portchannel del portchannel100
admin@r-qa-sw-eth-53:/tmp$ show interfaces portchannel
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
No.    Team Dev    Protocol    Ports
-----  ----------  ----------  -------
admin@r-qa-sw-eth-53:/tmp$
```
- New command output (if the output of a command-line utility has changed)
```
admin@r-qa-sw-eth-53:~$ sudo config portchannel add portchannel100
admin@r-qa-sw-eth-53:~$ sudo config portchannel member add portchannel100 Ethernet4
admin@r-qa-sw-eth-53:~$ sudo config portchannel member add portchannel100 Ethernet8
admin@r-qa-sw-eth-53:~$ show interfaces portchannel
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev        Protocol     Ports
-----  --------------  -----------  -------------------------
  100  PortChannel100  LACP(A)(Dw)  Ethernet8(D) Ethernet4(D)
admin@r-qa-sw-eth-53:~$ sudo config portchannel del portchannel100
Error: Portchannel portchannel100 contains members. Remove members before deleting Portchannel!
admin@r-qa-sw-eth-53:~$ show interfaces portchannel
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev        Protocol     Ports
-----  --------------  -----------  -------------------------
  100  PortChannel100  LACP(A)(Dw)  Ethernet8(D) Ethernet4(D)
admin@r-qa-sw-eth-53:~$ sudo config portchannel member del portchannel100 Ethernet4
admin@r-qa-sw-eth-53:~$ sudo config portchannel member del portchannel100 Ethernet8
admin@r-qa-sw-eth-53:~$ sudo config portchannel del portchannel100
admin@r-qa-sw-eth-53:~$ show interfaces portchannel
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
No.    Team Dev    Protocol    Ports
-----  ----------  ----------  -------
admin@r-qa-sw-eth-53:~$
```

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@madhanmellanox why do we get the usage when there is a specific error message due to the new checker?

File: config/main.py
Comment (liat-grozovik) [inline config/main.py:1234]: replace 'Portchannel' with 'Port channel'.

File: doc/Command-Reference.md
Comment (liat-grozovik) [inline doc/Command-Reference.md:4841]: replace 'Portchannel' with 'Port channel'.
IMO portchannel as CLI command, this is what we have as we dont have spaces. but for comments or CLI output messages i believe we should use it with space.

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

**Comment by madhanmellanox** [comment]
> @madhanmellanox why do we get the usage when there is a specific error message due to the new checker?
> are there any other UI which need to have this logic? should we add it also on orchagent while handling the port channel delete flow just in case it derives from config change ?

Addressed in the next commit.

File: doc/Command-Reference.md
Comment (madhanmellanox) [reply doc/Command-Reference.md]: Addressed in the next commit.

**Comment by madhanmellanox** [comment]
@yxieca @lguohan can someone from MSFT review this PR? I am waiting for approval from reviewers with write access.

**Comment by prsunny** [comment]
@madhanmellanox , Liat has "requested changes". Can you ask Liat for approval?

**Comment by madhanmellanox** [comment]
@liat-grozovik can you please approve this PR?

**Review by liat-grozovik — APPROVED** [review]

**Comment by madhanmellanox** [comment]
@prsunny can this PR be cherry picked to 201911?

**Comment by madhanmellanox** [comment]
@yxieca can this PR be cherry picked to 201911?

**Pull request merged**
